doit/cmd_base.py

```diff
@@ -310,30 +310,64 @@ def _load_from(cmd, namespace, cmd_list):
         return task_list, doit_config
 
 
-class ModuleTaskLoader(TaskLoader):
+class TaskLoader2(TaskLoader):
+    """Interface of task loaders with new-style API.
+
+    The default implementation assumes tasks are loaded from a namespace, mapping identifiers to
+    elements like functions (like task generators) or constants (like configuration values).
+
+    This API update separates the loading of the configuration and the loading of the actual tasks,
+    which enables additional elements to be available during task creation.
+    """
+    def __init__(self):
+        super().__init__()
+        self.namespace = None
+
+    def setup(self, opt_values):
+        """Delayed initialization.
+
+        To be implemented if the data is needed by derived classes.
+
+        :param opt_values: (dict) with values for cmd_options
+        """
+
+    def load_doit_config(self):
+        """Load doit configuration."""
+        return loader.load_doit_config(self.namespace)
+
+    def load_tasks(self, cmd, pos_args):
+        """Load tasks.
+
+        :return: (List[Task])
+        :param cmd: (doit.cmd_base.Command) current command being executed
+        :param pos_args: (list str) positional arguments from command line
+        """
+        return loader.load_tasks(self.namespace, self.cmd_names, cmd.execute_tasks)
+
+
+class ModuleTaskLoader(TaskLoader2):
     """load tasks from a module/dictionary containing task generators
     Usage: `ModuleTaskLoader(my_module)` or `ModuleTaskLoader(globals())`
     """
-    cmd_options = ()
-
     def __init__(self, mod_dict):
         super(ModuleTaskLoader, self).__init__()
```

Comment:
since we are touching this code, lets use modern ``super`` (without parameters)

Reply:
I usually don't touch existing code just for personal taste. PyCharm shows little inconsistencies to my code style settings everywhere, but as this might be personal, I don't try to convince. But sure, I'll adapt it for you.

Reply:
Yes, usually dont modify existing code. but this class is now a new class...

Which style inconsistencies? As far as I know I only deviate from PEP8  regarding blank lines.

Reply:
Don't worry, as I said, this is always personal taste and I of course accept the library's style.

If you are really interested I suggest to run a style checker (I use one built into PyCharm) and a linter like pylint.